core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ConnectionsGroup.java

```diff
@@ -14,11 +14,10 @@
 import java.util.concurrent.ConcurrentLinkedQueue;
 
 import org.eclipse.rdf4j.common.annotation.InternalUseOnly;
-import org.eclipse.rdf4j.common.transaction.TransactionSetting;
 import org.eclipse.rdf4j.sail.Sail;
 import org.eclipse.rdf4j.sail.SailConnection;
-import org.eclipse.rdf4j.sail.shacl.planNodes.BufferedSplitter;
-import org.eclipse.rdf4j.sail.shacl.planNodes.PlanNode;
+import org.eclipse.rdf4j.sail.shacl.abstractsyntaxtree.planNodes.BufferedSplitter;
```

Comment:
Two things:

1. let's call the package 'ast' rather than 'abstractsyntaxtree', to be consistent with how we do this for SPARQL and SeRQL as well.
2. I thought part of the idea was to have an AST that is not "just" usable by the SAIL, but also potentially for other SHACL implementations. If that is still the case it should be put in a in a more generic package: `org.eclipse.rdf4j.shacl.ast`, and ideally a separate module as well.

Reply:
It's called abstract syntax tree so that it can exist together with the old AST. We can rename it before merging.